src/main/scala/edu/arizona/sista/odin/impl/ThompsonVM.scala

```diff
@@ -19,9 +27,10 @@ object ThompsonVM {
   private case class SingleThread(
       tok: Int,
       inst: Inst,
+      dir: Direction,
```

Comment:
Each thread knows which direction it is matching (left-to-right or right-to-left)